src/certs/ca/cert/mod.rs

```diff
@@ -102,6 +106,35 @@ impl codicon::Encoder<Body> for Certificate {
     }
 }
 
+impl<'de> de::Deserialize<'de> for Certificate {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        use codicon::Decoder;
+
+        let bytes: Vec<u8> = de::Deserialize::deserialize(deserializer)?;
```

Comment:
```suggestion
        let bytes: Bytes<Vec<u8>> = de::Deserialize::deserialize(deserializer)?;
```

This might be the cause of your pain later. You are currently deserializing an array of `u8` which is (annoyingly) different than bytes. See the `serde_bytes` crate for details.